src/storage/src/hummock/local_version/pinned_version.rs

```diff
@@ -112,13 +118,41 @@ impl PinnedVersion {
             return None;
         }
         let version_id = version.id;
+        let t = std::mem::take(&mut version.table_change_log);
 
         Some(PinnedVersion {
-            version: Arc::new(version),
             guard: Arc::new(PinnedVersionGuard::new(
                 version_id,
                 self.guard.pinned_version_manager_tx.clone(),
             )),
+            table_change_log: Arc::new(RwLock::new(t)),
+            version: Arc::new(LocalHummockVersion::from(version)),
```

Comment:
I just want to leave a note here.

This `LocalHummockVersion::from` is an addtional `HummockVersion` conversion introduced by this PR. However I don't think it will have significant performance implications, as it primarily involves move semantics.